lib/rig/event_filter.ex

```diff
@@ -284,21 +285,23 @@ defmodule Rig.EventFilter do
         pid,
         {:refresh_subscriptions, subscriber, subscriptions, prev_subscriptions, done_callback}
       )
+
+      SubscriptionsMetrics.set_subscriptions(length(subscriptions))
     end
 
     :ok
   end
 
   # ---
 
-  @callback forward_event(CloudEvent.t()) :: :ok
-  @spec forward_event(CloudEvent.t()) :: :ok
-  def forward_event(%CloudEvent{} = event) do
+  @callback forward_event(CloudEvent.t(), String.t(), String.t()) :: :ok
+  @spec forward_event(CloudEvent.t(), String.t(), String.t()) :: :ok
+  def forward_event(%CloudEvent{} = event, source, topic) do
```

Comment:
Obviously you'd like to know `source` and `topic` for the metrics, but I'm worried about what this means in terms of semantics. The meaning of `forward_event` is (has been?): "take an event, forward it to any subscribers". To a RIG subscriber, `topic` doesn't mean anything, and also subscribing to an event does not depend on the source. I'd prefer `forward_event` to not take source and topic and arguments. Instead, I'd rather not include that in the metric. If the goal is counting events per Kafka topic, it should be counted in the Kafka handler.

Reply:
I'll take a look, my idea was to be able to monitor something like this (events):

Consumer:

- number of events consumed in total
- number of events that were skipped/ignored for some reason
- number of events that failed to be consumed
- number of events successfully forwarded to a subscription via whatever channel (ws, sse, lp)

Producer:

- number of successfully produced events
- number of events that failed to be produced

As a user when I want to create some dashboards I need some tags to be able to filter by some criteria:

- topic/namespace whatever we call it
- source - kafka, kinesis, etc.

These were my thoughts back then, but tbh have to go through it one more time and check what make sense as we changed a lot of thing in the meantime.